sklearn/utils/estimator_checks.py

```diff
@@ -1915,6 +1917,46 @@ def transform(self, X):
                     shallow_params.items()))
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_set_params(name, estimator_orig):
+    estimator = clone(estimator_orig)
+
+    # Trivial check to make sure set_params is working
+    params = estimator.get_params()
+    estimator.set_params(**params)
+
+    # Check that get_params() returns the same thing
+    # before and after set_params() with some fuzz
+    # values
+
+    test_values = [-np.inf, np.inf, None,
```

Comment:
I'd still consider making this list shorter, because it's about such a marginal thing that we can't catch perfectly anyyway. But calling `clone` and `get_params` fewer times below will improve runtime.